src/sentry/incidents/models.py

```diff
@@ -304,6 +316,9 @@ class AlertRule(Model):
     status = models.SmallIntegerField(default=AlertRuleStatus.PENDING.value)
     dataset = models.TextField()
     query = models.TextField()
+    environment = models.ManyToManyField(
```

Comment:
@mattrobenolt Do you know if it's safe to add a fk constraint on environment?

Reply:
Uhhh, maybe. We might wanna do this is_dangerous to make sure.

Reply:
@iProgramStuff I think this is probably safer as well. What you want to do here is split this pr into two pieces:
1. Just the migration, but you'll need to use
```
is_dangerous = True

operations = [
    migrations.SeparateDatabaseAndState(
        database_operations=<operations from your migration>,
        state_operations=[],
    )
]
```
This causes the database changes to be made, but doesn't update the actual state of the migrations. This means that when you merge it, other people won't end up with weird migration problems. You should then merge that pr, and work with someone on ops to run the migration manually (basically generate the ddl from the migration and give it to them to run). We're doing this just to be sure that we don't lock when trying to add the fk to environment.

2. Have the rest of this pr, including another migration which is kind of the complement of the other:
```
operations = [
    migrations.SeparateDatabaseAndState(
        database_operations=[],
        state_operations=<operations from your migration>,
    )
]
```
This updates the migration state to recognise that those models have been migrated, but doesn't actually generate any sql, since that will have been generated in the previous migration.

Reply:
This approach doesn't work, since having foreign keys from unknown tables causes problems in tests. So instead, what we're going to do is have #op run this DDL, then merge this PR after they're done.